contracts/FlowIDTableStaking.cdc

```diff
@@ -857,6 +857,70 @@ pub contract FlowIDTableStaking {
             }
         }
 
+        /// Enforces slot (count) limits for each node type (role). If more nodes
+        /// of a given type are registered than the limit, they are removed
+        /// randomly until the limits are met. When a node is removed, it's
+        /// committed tokens are transfered to it's unstaked bucket.
+        pub fun slotSelect() {
+            let allNodeIDs = FlowIDTableStaking.getNodeIDs()
+
+            let limits = [2, 2, 2, 2, 2]
+            var counts = [0, 0, 0, 0, 0]
```

Comment:
We would prefer the limits field to be stored as fields in the contract instead of hard-coded if possible. See how we added these "fields" in upgrades previous times for `stakingEnabled`.
https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowIDTableStaking.cdc#L1286

https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowIDTableStaking.cdc#L798-L802

Reply:
Also, wouldn't it be better to make these as dictionaries indexed by node role so we don't have the awkward off-by-one index?

Reply:
Firstly, thanks very much for the review! I'll change the data structure to a dictionary stored as a contract field. Would you then initialize it in the "init" function with default values as is done with stakingEnabled?

Reply:
yes, but you'll also need to make a method in the admin resource that initializes it in the account, because we'll need to run it to set it up after the contract update and if we ever want to change it

Reply:
I'm going to push changes shortly that will change the names to "slotLimits" and "nodeCounts" and change the arrays to dictionaries. The slotLimits will now be stored as a contract field as requested. I added a method to set them and I now use it in my test. I also assign default values.